brave/browser-laptop
# "New bookmark folder" and "New bookmark" buttons added to about:bookmarks

"New bookmark folder" and "New bookmark" buttons added to about:bookmarks

- sortableTable now allows you to provide a control instead of a string for the header
- about:bookmarks now properly sets tableID (not used yet)
- Adding a new bookmark now requires either a title or location (URL). Required because clicking new bookmark creates an empty object (versus existing buttons which always supply the current page).
- Fixed style issue with about:bookmarks (moved border-right on folderView to organizeView

Fixes https://github.com/brave/browser-laptop/issues/4684
Fixes https://github.com/brave/browser-laptop/issues/4866

Auditors: @jkup, @bbondy

## Test Plan
1. Launch Brave and visit about:bookmarks
2. Pick any folder on the left hand side and click it to select it
3. Click the +folder icon, above the folder view
4. Confirm that the new bookmark folder UI comes up and that it sets the parent folder properly
5. Click add and ensure folder gets added
6. Pick any folder and click it to select
7. In the "title" column, hit the star+ icon to add a new bookmark
8. Confirm you can't add an empty bookmark. Type and see the button enable itself. Delete chars and it disables itself again.
9. Confirm the parent folder is correct and type in a title/url. When done, hit save.
10. Confirm bookmark was added.

## Screenshots
**Two new buttons; folder with + and bookmark star with +**
![screen shot 2016-10-18 at 1 22 49 am](https://cloud.githubusercontent.com/assets/4733304/19469766/82a36ed6-94d1-11e6-8d94-da1b98a4f8f9.png)

**Example shot of clicking the add folder button**
![screen shot 2016-10-18 at 1 21 52 am](https://cloud.githubusercontent.com/assets/4733304/19469768/86870b0c-94d1-11e6-8b57-8869f99f6da4.png)
